app/controllers/course/learning_map_controller.rb

```diff
@@ -87,29 +91,36 @@ def map_conditionals_to_nodes
     generate_node_depths(nodes)
   end
 
-  def generate_all_node_relations
+  def generate_all_node_relations # rubocop:disable Metrics/AbcSize
     relations = init_all_node_relations
     node_ids_to_children = relations[:node_ids_to_children]
     node_ids_to_parents = relations[:node_ids_to_parents]
+    node_ids_to_unlock_level = relations[:node_ids_to_unlock_level]
 
     @conditionals.each do |conditional|
       node_id = get_node_id(conditional)
 
       conditional.conditions.each do |condition|
-        next if condition.actable_type == Course::Condition::Level.name
+        if condition.actable_type == Course::Condition::Level.name
+          level_condition = Course::Condition::Level.find(condition.actable_id)
+          node_ids_to_unlock_level[node_id] = level_condition.minimum_level
+          next
+        end
 
         parent = map_condition_to_parent(condition)
         node_ids_to_children[parent[:id]].push({ id: node_id, is_satisfied: parent[:is_satisfied] })
         node_ids_to_parents[node_id].push(parent)
       end
     end
 
-    { node_ids_to_children: node_ids_to_children, node_ids_to_parents: node_ids_to_parents }
+    { node_ids_to_children: node_ids_to_children, node_ids_to_parents: node_ids_to_parents,
+      node_ids_to_unlock_level: node_ids_to_unlock_level }
   end
 
   def init_all_node_relations
     { node_ids_to_children: @conditionals.map { |conditional| [get_node_id(conditional), []] }.to_h,
```

Comment:
Btw just a minor nit - I think rubocop doesn't like it when we prefix our methods with `get`, from what I know.

Reply:
I think this rule only applies to accessor/getter methods with 0 args instead of the way that I defined my methods to retrieve data based on the args being passed in:
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/AccessorMethodName

Reply:
Roger, thanks for clarifying!